x-pack/plugins/observability_ai_assistant/public/functions/get_dataset_info.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Serializable } from '@kbn/utility-types';
+import { RegisterFunctionDefinition } from '../../common/types';
+import type { ObservabilityAIAssistantService } from '../types';
+
+export function registerGetDatasetInfoFunction({
+  service,
+  registerFunction,
+}: {
+  service: ObservabilityAIAssistantService;
+  registerFunction: RegisterFunctionDefinition;
+}) {
+  registerFunction(
+    {
+      name: 'get_dataset_info',
+      contexts: ['core'],
+      description: `Use this function to get information about indices/datasets available and the fields available on them.
+
+      providing empty string as index name will retrieve all indices
+      if index is provided and it doesnt match any indices in elasticsearch list of all indices will be provided as well
+      else list of all fields for the given index will be given. if no fields are returned this means no indices were matched by provided index pattern.
+      wildcards can be part of index name.`,
+      descriptionForUser:
```

Comment:
I don't think this is useful for the user. WDYT about hiding it for users, but still making it available to the LLM? You can use FunctionVisibility.System for this.